File: Vesti/Vesti/ArticleDetailsViewController.m
//
//  ArticleDetailsViewController.m
//  Vesti
//
//  Created by Djuro Alfirevic on 4/26/16.
//  Copyright © 2016 Djuro Alfirevic. All rights reserved.
//

#import "ArticleDetailsViewController.h"
#import "WebViewController.h"
#import "AsyncImageView.h"

@interface ArticleDetailsViewController()
@property (weak, nonatomic) IBOutlet AsyncImageView *asyncImageView;
@property (weak, nonatomic) IBOutlet UILabel *titleLabel;
@property (weak, nonatomic) IBOutlet UITextView *descTextView;
@end

@implementation ArticleDetailsViewController

#pragma mark - View lifecycle

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.asyncImageView.imageURL = self.article.imageURL;
    self.titleLabel.text = self.article.title;
    self.titleLabel.adjustsFontSizeToFitWidth = YES;
    self.descTextView.text = self.article.desc;
}

#pragma mark - Segue management

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([segue.identifier isEqualToString:@"WebSegue"]) {
        WebViewController *toViewController = segue.destinationViewController;
        toViewController.article = self.article;
    }
}

@end

File: VestiS/VestiS/ArticleDetailsViewController.swift
//
//  ArticleDetailsViewController.swift
//  VestiS
//
//  Created by Djuro Alfirevic on 8/3/16.
//  Copyright © 2016 Djuro Alfirevic. All rights reserved.
//

import UIKit

class ArticleDetailsViewController: UIViewController {

    // MARK: - Properties
    @IBOutlet weak var asyncImageView: AsyncImageView!
    @IBOutlet weak var titleLabel: UILabel!
    @IBOutlet weak var descTextView: UITextView!
    var article: Article?
    
    // MARK: - View lifecycle
    override func viewDidLoad() {
        super.viewDidLoad()
        
        if let article = self.article {
            self.asyncImageView.imageURL = article.imageURL
            self.titleLabel.text = article.title
            self.titleLabel.adjustsFontSizeToFitWidth = true
            self.descTextView.text = article.desc
        }
    }
    
    // MARK: - Segue
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        if segue.identifier == "WebSegue" {
            let toViewController = segue.destinationViewController as! WebViewController
            toViewController.article = self.article
        }
    }
    
}